plugins/chat/assets/javascripts/discourse/components/chat-channel-members.gjs

```diff
@@ -0,0 +1,125 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import ChatUserInfo from "discourse/plugins/chat/discourse/components/chat-user-info";
+import gt from "truth-helpers/helpers/gt";
+import { cached, tracked } from "@glimmer/tracking";
+import { INPUT_DELAY } from "discourse-common/config/environment";
+import discourseDebounce from "discourse-common/lib/debounce";
+import { modifier } from "ember-modifier";
+import isElementInViewport from "discourse/lib/is-element-in-viewport";
+import DcFilterInput from "discourse/plugins/chat/discourse/components/dc-filter-input";
+import I18n from "I18n";
+import { hash } from "@ember/helper";
+import { schedule } from "@ember/runloop";
+
+export default class ChatChannelMembers extends Component {
+  <template>
+    {{! template-lint-disable modifier-name-case }}
+    <div class="chat-channel-members">
+      <DcFilterInput
+        @class="chat-channel-members__filter"
+        @filterAction={{this.mutFilter}}
+        @icons={{hash right="search"}}
+        placeholder={{this.filterPlaceholder}}
+        {{this.focusInput}}
+      />
+
+      {{#if (gt @channel.membershipsCount 0)}}
+        <ul class="chat-channel-members__list" {{this.fill}}>
+          {{#each this.members as |membership|}}
+            <li class="chat-channel-members__list-item">
+              <ChatUserInfo @user={{membership.user}} @avatarSize="tiny" />
+            </li>
+          {{else}}
+            {{#if this.noResults}}
+              <li
+                class="chat-channel-members__list-item -no-results alert alert-info"
+              >
+                {{this.noMembershipsFoundLabel}}
+              </li>
+            {{/if}}
+          {{/each}}
+        </ul>
+
+        <div {{this.loadMore}}>
+          <br />
+        </div>
+      {{else}}
+        <p class="alert alert-info">
+          {{this.noMembershipsLabel}}
+        </p>
+      {{/if}}
+    </div>
+  </template>
+
+  @service chatApi;
+  @service modal;
+  @service loadingSlider;
+
+  @tracked filter = "";
+
+  filterPlaceholder = I18n.t("chat.members_view.filter_placeholder");
+  noMembershipsFoundLabel = I18n.t("chat.channel.no_memberships_found");
+  noMembershipsLabel = I18n.t("chat.channel.no_memberships");
+
+  focusInput = modifier((element) => {
+    schedule("afterRender", () => {
+      element.focus();
+    });
+  });
+
+  fill = modifier((element) => {
+    this.resizeObserver = new ResizeObserver(() => {
+      if (isElementInViewport(element)) {
+        this.load();
+      }
+    });
+
+    this.resizeObserver.observe(element);
+
+    return () => {
+      this.resizeObserver.disconnect();
+    };
+  });
+
+  loadMore = modifier((element) => {
+    this.intersectionObserver = new IntersectionObserver(this.load);
+    this.intersectionObserver.observe(element);
+
+    return () => {
+      this.intersectionObserver.disconnect();
+    };
+  });
```

Comment:
It's an efficient solution to auto fill and load more:
- if the container is not taking full height of the page, it will try to load more
- when the element at the bottom of the page becomes visible, it will try to load more